app/controllers/api/v0/health_checks_controller.rb

```diff
@@ -0,0 +1,37 @@
+module Api
+  module V0
+    class HealthChecksController < ApiController
+      before_action :authenticate_with_token
```

Comment:
I'd favor a standard HTTP way to do this, using Rails's [ActionController::HttpAuthentication::Token](https://api.rubyonrails.org/classes/ActionController/HttpAuthentication/Token.html) method:

```ruby
def authenticate
  authenticate_or_request_with_http_token do |token, options|
    # Compare the tokens in a time-constant manner, to mitigate
    # timing attacks.
    ActiveSupport::SecurityUtils.secure_compare(token, TOKEN)
  end
end
```

`ActiveSupport::SecurityUtils.secure_compare` is important, we use it for the `api-key` authentication as well.

This way the client could just be:

`curl -H "Authorization: Token TOKEN_VALUE" https://dev.to/api/healt-checks/app`

Reply:
For sensitive endpoints, I think that would be more important but since these are simple health check endpoints I think the simple comparison will suffice.